src/androidTest/java/com/nutomic/syncthingandroid/test/syncthing/AppConfigReceiverTest.java

```
@@ -0,0 +1,169 @@
+     * In this case starting the service is allowed
+     */
+    public void testStartSyncthingServiceBackground() {
+        PreferenceManager.getDefaultSharedPreferences(mContext)
+                .edit()
+                .putBoolean(SyncthingService.PREF_ALWAYS_RUN_IN_BACKGROUND, true)
+                .commit();
+
+        Intent intent = new Intent(new Intent(mContext, AppConfigReceiver.class));
+        intent.setAction(AppConfigReceiver.ACTION_START);
+
+        mReceiver.onReceive(mContext, intent);
+        assertEquals("Start SyncthingService Background", 1, mContext.getReceivedIntents().size());
+        assertEquals("Start SyncthingService Background", SyncthingService.class.getName(), mContext.getReceivedIntents().get(0).getComponent().getClassName());
+    }
+
+    /**
+     * Method #1:
+     * Test stopping the service if "alway run in background" is enabled.
+     * In this mode stopping the service is not allowed. In that case a notification is created.
+     * (This crashes in this test context which indicates that the test succeeded.)
+     */
+    public void testStopSyncthingServiceBackground() {
+        PreferenceManager.getDefaultSharedPreferences(mContext)
+                .edit()
+                .putBoolean(SyncthingService.PREF_ALWAYS_RUN_IN_BACKGROUND, true)
+                .commit();
+
+        Intent intent = new Intent(new Intent(mContext, AppConfigReceiver.class));
+        intent.setAction(AppConfigReceiver.ACTION_STOP);
+
+        // onReceive now tries to create a notification.
+        // This is expected to throw an exception as creating a notification is not supported in this Test-Context
+        // so an exception is an indicator of success in this case.
+        try {
+            mReceiver.onReceive(mContext, intent);
+            Assert.fail("Stop SyncthingService Background (1) - didn't fail to create the notification");
+
+        } catch (NullPointerException e) {
```

I really don't like to catch NPE like this, because it can easily hide actual errors. 

I'm not sure why this crash happens. The error is somewhere in the call to `PendingIntent.getActivity()`, but I don't see any reason for that. But instead of catching the NPE, you can override `getContentResolver()` in MockContext, and throw a custom exception (or set a flag).